### ISSUE TITLE
Interpretation of TF Activity

### ISSUE THREAD

**Issue by lizzyjoan** [saezlab/decoupleR#96]
Hello,
First off- thank you so much for the excellent package.
I had some confusion though about the interpretation of TF activity, with regards to activator/repressor vs activation/inactivation. I see from the example in the deboupleR vignette "Get started" tab, the TF activity is incorrectly assigned for "T3" when using the gsea method vs the ulm method (due to gsea not taking weights into account unlike ulm.) So, in this example, because a TF was mostly repressing, it was considered "active" because the expression of those repressor targets was low (even though it's mostly a repressor, not activator, it's "active" -- is that correct?)
Then, if I've understood that correctly -- would this be the case as well for mlm (since it also takes weights into account), as long as weights are included with inputs? Or in any case, is TF activity showing a TF is acting more as a repressor or activator?

In other words -- how should a negative TF activity using mlm be interpreted? A) That TF is acting as a repressor in that context, or B) that the TF is just regulating less altogether (regardless of its specific activation/repression TF-gene interactions)?

Thank you for your time and hope this makes sense!

**Comment by PauBadiaM** [comment]
Hi @lizzyjoan,

First of all sorry for the late reply, I had replied to your issue but I think I forgot to save it.
A negative activity for a given TF obtained by `mlm` (or any other weighted method) can be explained by two gene patterns, either that the genes it has a repressive role are up-regulated (the inhibition did not work), or that the genes it has an activator role are down-regulated (the activation did not work). The only way to explore this is to plot the expression levels of the TF's target genes. In this [vignette](https://saezlab.github.io/decoupleR/articles/tf_bk.html#visualization), at the end we visualize the targets to better understand why is it active:
![](https://saezlab.github.io/decoupleR/articles/tf_bk_files/figure-html/targets-1.png)
As you can see, all positive genes are upregulated, and all negative genes are downregulated, meaning that the TF is active. Hope this is helpful! Let me know if you have more questions.

**Comment by lizzyjoan** [comment]
Thank you, that was very helpful!
